### PR TITLE
Consume PF4 pagination from core

### PR DESCRIPTION
**Pull request by stejskalleos**

### CODE REVIEW

**Review by ezr-ondrej — CHANGES_REQUESTED** [review]
I did not check, but the properties passed to Pagination here seems same as in REX, so I believe we have the same issues as mentioned in https://github.com/theforeman/foreman_remote_execution/pull/682#pullrequestreview-856017528

**Comment by stejskalleos** [comment]
Waiting for https://github.com/theforeman/foreman_remote_execution/pull/682

**Comment by stejskalleos** [comment]
Updated, removed unused props and fixed `per_page` args

File: webpack/components/PreupgradeReportsList/index.js [hunk 1]
```diff
@@ -44,9 +44,9 @@ const PreupgradeReportsList = ({
       <Pagination
         viewType="list"
         itemCount={allEntries.length}
-        pagination={pagination}
+        // pagination={pagination}
         onChange={newPage => setPagination({ ...pagination, ...newPage })}
-        dropdownButtonId="preupgrade-report-entries-pagination-dropdown"
+        // dropdownButtonId="preupgrade-report-entries-pagination-dropdown"
```
Comment (Ron-Lavi) [inline webpack/components/PreupgradeReportsList/index.js:49]: the commented lines can be removed

**Review by Ron-Lavi — COMMENTED** [review]
Overall looks great, I don't have a pre-upgrade report so couldn't actually test it,
@ezr-ondrej do you have the setup?

**Comment by stejskalleos** [comment]
> Overall looks great, I don't have a pre-upgrade report so couldn't actually test it

You can generate report pretty simple:

Install Leapp on CentOS machine:
```
yum install -y yum-utils
yum-config-manager --add-repo https://copr.fedorainfracloud.org/coprs/g/oamg/leapp/repo/epel-7/group_oamg-leapp-epel-7.repo
yum install leapp
```
run preupgrade job in Foreman
If you don't have much records in report, you can add some more:
```
bundle exec rails console
100.times { PreupgradeReportEntry.first.dup.save }
```

**Comment by Ron-Lavi** [comment]
Having some issues on my machine, @ezr-ondrej mind taking a look?

**Review by ezr-ondrej — APPROVED** [review]
It was quite hard to setup (leapp is not very straight forward to install).
It could be beneficial to have some script to feed the DB with dummy records for ppl to test out the plugin without having the production setup working. Or at least a documentation on how to set it up :)
Tho this works :+1:  Thanks @stejskalleos

**Comment by ezr-ondrej** [comment]
We need someone with merge permissions tho.

**Comment by Ron-Lavi** [comment]
Maybe @adamruzicka for the rescue? :)

**Comment by adamruzicka** [comment]
Shouldn't this also bump the required foreman version in engine.rb?

**Comment by stejskalleos** [comment]
> Shouldn't this also bump the required foreman version in engine.rb?

Yop, updated to `3.2`

**Review by adamruzicka — APPROVED** [review]
LGTM

**Comment by adamruzicka** [comment]
Thank you everyone!

**Pull request merged**